libnavigation-core/src/test/java/com/mapbox/navigation/core/lifecycle/RequireMapboxNavigationDelegateTest.kt

```diff
@@ -0,0 +1,345 @@
+package com.mapbox.navigation.core.lifecycle
+
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.core.MapboxNavigation
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.unmockkAll
+import io.mockk.verify
+import io.mockk.verifyOrder
+import org.junit.After
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class RequireMapboxNavigationDelegateTest {
+
+    private class SystemUnderTest(
+        onCreatedObserver: MapboxNavigationObserver? = null,
+        onStartedObserver: MapboxNavigationObserver? = null,
+        onResumedObserver: MapboxNavigationObserver? = null,
+        onInitialize: (() -> Unit)? = null
+    ) : LifecycleOwner {
+        private val lifecycleRegistry = LifecycleRegistry(this)
+            .also { it.currentState = Lifecycle.State.INITIALIZED }
+
+        override fun getLifecycle(): Lifecycle = lifecycleRegistry
+
+        fun moveToState(state: Lifecycle.State) {
+            lifecycleRegistry.currentState = state
+        }
+
+        val mapboxNavigation by RequireMapboxNavigationDelegate(
+            onCreatedObserver = onCreatedObserver,
+            onStartedObserver = onStartedObserver,
+            onResumedObserver = onResumedObserver,
+            onInitialize = onInitialize
+        )
+    }
+
+    @Before
+    fun setup() {
+        mockkObject(MapboxNavigationApp)
+    }
+
+    @After
+    fun teardown() {
+        unmockkAll()
+    }
+
+    @Test(expected = IllegalStateException::class)
+    fun `crashes before setup`() {
+        SystemUnderTest().mapboxNavigation
+    }
+
+    @Test
+    fun `mapboxNavigation is available when current is not null`() {
+        every { MapboxNavigationApp.current() } returns mockk()
+
+        SystemUnderTest().mapboxNavigation
+    }
+
+    @Test
+    fun `mapboxNavigation is available after MapboxNavigationApp#setup and lifecycle is CREATED`() {
```

Comment:
Don't you also want to test STARTED, RESUMED, PAUSED and STOPPED?

Reply:
All the lifecycles are tested as separate unit tests. This unit test is specific to the `CREATED` lifecycle because this is the minimum setup to not crash